File: Data Structures & Algorithms/two-integer-sum/submission-1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        prevMap = {}
        for i,x in enumerate(nums):
            dif = target - x
            if dif in prevMap:
                return [prevMap[dif], i]
            prevMap[x] = i
        return

File: Data Structures & Algorithms/binary-search/submission-1.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int n = nums.size();
        int st = 0, dr = n - 1;
        while (st <= dr) {
            int mid = st + (dr - st) / 2;
            if (nums[mid] == target) return mid;
            if (nums[mid] < target) st = mid + 1;
            else dr = mid - 1;
        }
        return -1;
    }
};
